docs/code/IDataViewTypeSystem.md

```diff
@@ -0,0 +1,844 @@
+leads to some nice invariants. For example, when loading a text file with
+sparse row specifications, it's desirable for the result to be the same
+whether the row is first processed entirely as `TX` values, then parsed, or
+processed directly into numeric values, that is, parsing as the row is
+processed. In the latter case, it is simple to map implicit items (suppressed
+due to sparsity) to zero. In the former case, these items are first mapped to
+the empty text value. To get the same result, we need empty text to map to
+zero.
+
+### Floating Point
+
+There are standard conversions from `R4` to `R8` and from `R8` to `R4`. These
+are the standard IEEE 754 conversions (using unbiased round-to-nearest in the
+case of `R8` to `R4`).
+
+### Signed Integer
+
+There are standard conversions from each signed integer type to each other
+signed integer type. These conversions map `NA` to `NA`, map any other numeric
+value that fits in the destination type to the corresponding value, and maps
+any numeric value that does not fit in the destination type to `NA`. For
+example, when mapping from `I1` to `I2`, the source `NA` value, namely 0x80,
+is mapped to the destination `NA` value, namely 0x8000, and all other numeric
+values are mapped as expected. When mapping from `I2` to `I1`, any value that
+is too large in magnitude to fit in `I1`, such as 312, is mapped to `NA`,
+namely 0x80.
+
+### Signed Integer to Floating Point
+
+There are standard conversions from each signed integer type to each floating-
+point type. These conversions map `NA` to `NA`, and map all other values
+according to the IEEE 754 specification using unbiased round-to-nearest.
+
+### Unsigned Integer
+
+There are standard conversions from each unsigned integer type to each other
+unsigned integer type. These conversions map any numeric value that fits in
+the destination type to the corresponding value, and maps any numeric value
+that does not fit in the destination type to zero. For example, when mapping
+from `U2` to `U1`, any value that is too large in magnitude to fit in `U1`,
+such as 312, is mapped to zero.
+
+### Unsigned Integer to Floating Point
+
+There are standard conversions from each unsigned integer type to each
+floating-point type. These conversions map all values according to the IEEE
+754 specification using unbiased round-to-nearest.
+
+### Key Types
+
+There are standard conversions from one key type to another, provided:
+
+* The source and destination key types have the same `Min` and `Count` values.
+
+* Either the number of bytes in the destination's underlying type is greater
+  than the number of bytes in the source's underlying type, or the `Count`
+  value is positive.  In the latter case, the `Count` is necessarily less than
+  `2^^k`, where `k` is the number of bits in the destination type's underlying
+  type. For example, `U1[1-*] `can be converted to `U2[1-*]`, but `U2[1-*]`
```

Comment:
```
`U1[1-*] `can be converted
```
 has the space on the wrong side of the back tick.

Reply:
When I first read this, my brain somehow filled in "traintrack" for "back tick." I imagined a poor little space, on the wrong side of the backtick, suddenly finding itself set upon by the unsavory characters in the KeyType gang...

---
In reply to: [188813760](https://github.com/dotnet/machinelearning/pull/173#discussion_r188813760) [](ancestors = 188813760)

Reply:
I like how you think...

Reply:
Hi Justin, just clarifying, was there a comment attached here? I've reviewed the language in the section about up-conversion, it *seems* correct?

---
In reply to: [188877915](https://github.com/dotnet/machinelearning/pull/173#discussion_r188877915) [](ancestors = 188877915)

Reply:
Oh oh oh... this was a comment on the earlier issue. Geez CodeFlow github integration sure isn't great at tracking replies. :P

---
In reply to: [188963444](https://github.com/dotnet/machinelearning/pull/173#discussion_r188963444) [](ancestors = 188963444,188877915)